### PR TITLE
Adds python code for HLLC

### PR DESCRIPTION
**Pull request by bishtgautam**
Adds python code developed by Ashwin Raman <araman@crimson.ua.edu> for HLLC solver on structured grids.

### CODE REVIEW

File: python/2d_structured_hllc_roe.py [hunk 1]
```diff
@@ -0,0 +1,1174 @@
...
+from exact_riemann import wetbed
+from scipy.optimize import fsolve
+from scipy.integrate import solve_ivp
+
+
+C=0.5
+n = 0.0 #manning friction
+tend = 60.0
+g = 9.81;
+htol = 1.0e-2 #height tolerance below which hl&hr ; ul&ur is set to zero
```
Comment (donghuix) [inline python/2d_structured_hllc_roe.py:15]: Is this the ``tiny_h`` in ex2b.c? I wonder if a smaller value should be used as threshold here. Or is this threshold required by HLLC solver? In my understanding, 1cm depth of water is not negligible.

File: python/2d_structured_hllc_roe.py [hunk 2]
```diff
@@ -0,0 +1,1174 @@
...
+                    ur = 0.0;
+                    vr = 0.0;
+                else:
+                    ur = qxR/hr;
+                    vr = qyR/hr;
+                
+                al = np.sqrt(g*hl)
+                ar = np.sqrt(g*hr)
+                
+                if solver == 'hllc':
```
Comment (donghuix) [inline python/2d_structured_hllc_roe.py:855]: Why not write a function for the hllc solver and call the function here?

File: python/2d_structured_hllc_roe.py [hunk 3]
```diff
@@ -0,0 +1,1174 @@
...
+
+# bslope = ['flatbed','slope_upwards','slope_downwards','obstruction',
+#           'bump','flatbed+sloping_stage','2d_parabolic_dam','riemann_problem',
+#           '2d_dambreak','circular_dambreak','bump_steady_flow','bump_partial_cover']
+
+# choose the problem you need from the list of bslope below by setting ibslope
+# to the correct index
+bslope = ['bump','2d_parabolic_dam','riemann_problem',
+          '2d_dambreak','circular_dambreak','bump_steady_flow','bump_partial_cover']
+ibslope = 4
```
Comment (donghuix) [inline python/2d_structured_hllc_roe.py:29]: I suggest to pass the option from command line arguments. This also applies to the following ibump, ibc,.... I can change it after the code is merged.

**Review by donghuix — APPROVED** [review]
Some comments for you to consider.

**Comment by bishtgautam** [comment]
@donghuix Thanks for your comments. I will merge this PR as is to keep track of the exact version that Ashwin as shared with us. We can make updates in a future PR.

**Pull request merged**